docs/DeveloperGuide.adoc

```diff
@@ -798,51 +798,158 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*` |prisoner |know my release date |look forward to the day I can see my family/friends again
+
+|`*` |police officer |see my assigned prisoners’ social networks: friends, enemies, family, etc… |use this as reference when dealing with inmates
+
+|`*` |police officer |assign prisoners to solitary confinement |punish those that have continuously committed serious offences
+
+|`*` |cook |check dietary requirements of all people in prison |cook sufficient food
+
+|`*` |counselor |check what prisoners' crimes |understand my patients better
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
 
 _{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `AddressBook` (AB) unless specified otherwise)
+
+[discrete]
+=== Use case: Add a prisoner
+Actor: police officer (PO)
+
+*MSS*
+
+1.  PO requests to add prisoner
+2.  AB adds prisoner
+3.  PO requests to list cells that are empty/not filled
+4.  AB shows a list of empty/not filled cells
+5.  PO requests to view a specific cell’s prisoner’s details
+6.  AB shows details of prisoner in cell
+7.  PO requests to add a prisoner to a specific cell
+8.  AB adds prisoner to cell
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There are no cells available for new prisoner.
++
+[none]
+** 2a1. AB shows an error message.
++
+Use case ends.
+
+* 5a. The given index is invalid, outside range.
++
+[none]
+** 5a1. AddressBook shows an error message.
++
+Use case resumes at step 4.
+
+* 5b. The given index of cell has no prisoners.
++
+[none]
+** 5b1. AddressBook shows an error message.
++
+Use case resumes at step 4.
+
+* 7a. The given index is invalid.
++
+[none]
+** 7a1. AddressBook shows an error message.
++
+Use case resumes at step 6.
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Assign prisoner to a cell
+Actor: police officer (PO)
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  PO selects a prisoner
+2.  PO requests a list of available cells with room available
+3.  AB shows a list of available cells
+4.  PO requests to add prisoner to vacant cell
+5.  AB updates prisoner status and cell status
 +
 Use case ends.
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+* 2a. The cell list is empty.
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. Prisoner is already assigned to a cell.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 3a1. AB removes cell tag from prisoner status and removes prisoner from cell status
 +
 Use case resumes at step 2.
+[discrete]
+=== Use case: Get the list of prisoners in a certain cell
```

Comment:
The rendering of this guide shows an indentation error at this point. Please try to fix that.